apps/web/src/actions/runs/stopByDocument.ts

```diff
@@ -0,0 +1,35 @@
+'use server'
+
+import { z } from 'zod'
+import { getRunByDocument } from '@latitude-data/core/services/runs/active/byDocument/get'
+import { stopRunByDocument } from '@latitude-data/core/services/runs/active/byDocument/stop'
+import { withProject, withProjectSchema } from '../procedures'
+
+/**
+ * Stops a run using document-scoped storage.
+ * This is the document-scoped version of stopRunAction.
+ */
+export const stopRunByDocumentAction = withProject
+  .inputSchema(
+    withProjectSchema.extend({
+      documentUuid: z.string(),
+      runUuid: z.string(),
+    }),
+  )
+  .action(async ({ parsedInput, ctx }) => {
+    const run = await getRunByDocument({
```

Comment:
so you've reimplemented the same services but with byDocument* suffix, do we need to keep the old services? also the apis are weird, both receiving a run uuid and a document uuid what is gonna happen? it's gonna find by uuid and then check it matches the document uuid you passed? if you already have the run uuid why do you need to pass the document uuid?